### PR TITLE
PB-531: Improve report a problem error management for big files

### PR DESCRIPTION
**Pull request by ltshb**
The backend only support attachement up 10MB, above it returns a 413 error.
Same for kml it only support kml up to 2MB.

So do the validation up front and provide a nice message to the user.

image

[Test link](https://sys-map.dev.bgdi.ch/preview/bug-pb-531-report-problem-file-limit/index.html)

### CODE REVIEW

File: src/utils/components/FileInput.vue [hunk 1]
```diff
@@ -265,7 +267,10 @@ function onFileSelected(evt) {
                 data-cy="file-input-invalid-feedback"
             >
                 {{
-                    i18n.t(invalidMessage).replace('ALLOWED_FORMATS', acceptedFileTypes.join(', '))
+                    i18n
+                        .t(invalidMessage)
+                        .replace('ALLOWED_FORMATS', acceptedFileTypes.join(', '))
+                        .replace('MAX_FILE_SIZE', maxFileSizeHuman)
```
Comment (pakb) [inline src/utils/components/FileInput.vue:273]: might make sense to use some of our I18N tool here instead
https://vue-i18n.intlify.dev/guide/essentials/syntax.html#interpolations

File: src/utils/utils.js [hunk 2]
```diff
@@ -240,3 +240,14 @@ export function isValidEmail(email) {
...
 }
+
+/**
+ * Human readable size
+ *
+ * @param {Number} size Size in bytes
+ * @returns {String} Human readable size
+ */
+export function humanFileSize(size) {
+    let i = size == 0 ? 0 : Math.floor(Math.log(size) / Math.log(1024))
```
Comment (pakb) [inline src/utils/utils.js:251]: I don't think you are changing `i` value down the line, could be a `const` then

File: src/utils/utils.js [hunk 3]
```diff
@@ -240,3 +240,14 @@ export function isValidEmail(email) {
...
+    return +(size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i]
```
Comment (pakb) [inline src/utils/utils.js:252]: what is this `+` doing at the start of the expression?

Comment (ltshb) [reply src/utils/utils.js:252]: Copy paste from the net 😉 I'll remove it

File: src/utils/components/FileInput.vue (hunk 1 above)
Comment (ltshb) [reply src/utils/components/FileInput.vue:273]: good idea

**Review by hansmannj — APPROVED** [review]
Thanks for adapting validation and throwing a detailed error message!
LGTM :+1:

File: src/modules/i18n/locales/de.json [hunk 4]
```diff
@@ -199,7 +199,7 @@
     "feedback_success_message": "Danke für Ihre Nachricht. Diese wurde erfolgreich übermittelt.",
     "feedback_unsupported_format": "Dieser Dateityp wird leider nicht unterstützt. Bitte verwenden Sie eine .pdf, .zip, .jpg, .jpeg, .kml, .kmz oder .gpx Datei.",
     "file_is_not_kml": "Dieses File ist keine KML Datei. ",
-    "file_too_large": "Die Datei ist zu gross.",
+    "file_too_large": "Die Datei ist zu gross (maximal erlaubte Größe: MAX_FILE_SIZE).",
```
[the remark this answers is not in the document]
Comment (ltshb) [reply src/modules/i18n/locales/de.json:202]: done

**Review by pakb — APPROVED** [review]

**Pull request merged**